tests/storage/storage.go

```diff
@@ -965,14 +967,24 @@ var _ = SIGDescribe("Storage", func() {
 						}
 						return k8sv1.ConditionFalse
 					}, 30, 1).Should(Equal(k8sv1.ConditionTrue))
+					pod, err = ThisPod(pod)()
+					Expect(err).ToNot(HaveOccurred())
 
 					By("Determining the size of the mounted directory")
 					diskSizeStr, _, err := tests.ExecuteCommandOnPodV2(virtClient, pod, pod.Spec.Containers[0].Name, []string{"/usr/bin/bash", "-c", fmt.Sprintf("df %s | tail -n 1 | awk '{print $4}'", mountDir)})
 					Expect(err).ToNot(HaveOccurred())
 					diskSize, err = strconv.Atoi(strings.TrimSpace(diskSizeStr))
 					diskSize = diskSize * 1000 // byte to kilobyte
 					Expect(err).ToNot(HaveOccurred())
+				})
 
+				AfterEach(func() {
+					if vmi != nil {
```

Comment:
Btw. here you could also have used `Eventually(AllVMIs(NamespaceTestDefault), 120).Should(BeGone())`